lib/shelley/src/Cardano/Wallet/Shelley/Logging.hs

```diff
@@ -0,0 +1,338 @@
+        Nothing -> const nullTracer
+        Just s -> filterSeverity (const $ pure s)
+
+    mkTrace
+        :: (HasPrivacyAnnotation a, HasSeverityAnnotation a, ToText a)
+        => (Tracers' (Const Text) -> Const Text a)
+        -> Trace IO Text
+        -> Tracer IO a
+    mkTrace f =
+        trMessageText . appendName (getConst $ f tracerLabels)
+
+-- | Strings that the user can refer to tracers by.
+tracerLabels :: Tracers' (Const Text)
+tracerLabels = Tracers
+    { applicationTracer  = Const "application"
+    , apiServerTracer    = Const "api-server"
+    , walletEngineTracer = Const "wallet-engine"
+    , walletDbTracer     = Const "wallet-db"
+    , poolsEngineTracer  = Const "pools-engine"
+    , poolsDbTracer      = Const "pools-db"
+    , ntpClientTracer    = Const "ntp-client"
+    , networkTracer      = Const "network"
+    }
+
+-- | Names and descriptions of the tracers, for user documentation.
+tracerDescriptions :: [(String, String)]
+tracerDescriptions =
+    [ ( lbl applicationTracer
+      , "About start-up logic and the server's surroundings."
+      )
+    , ( lbl apiServerTracer
+      , "About the HTTP API requests and responses."
+      )
+    , ( lbl walletEngineTracer
+      , "About background wallet workers events and core wallet engine."
+      )
+    , ( lbl walletDbTracer
+      , "About database operations of each wallet."
+      )
+    , ( lbl poolsEngineTracer
+      , "About the background worker monitoring stake pools and stake pools engine."
+      )
+    , ( lbl poolsDbTracer
+      , "About database operations on stake pools."
+      )
+    , ( lbl ntpClientTracer
+      , "About ntp-client."
+      )
+    , ( lbl networkTracer
+      , "About network communication with the node."
+      )
+    ]
+  where
+    lbl f = T.unpack . getConst . f $ tracerLabels
+
+-- | Use a 'nullTracer' for each of the 'Tracer's in 'Tracers'
+nullTracers :: Monad m => Tracers m
+nullTracers = Tracers
+    { applicationTracer  = nullTracer
+    , apiServerTracer    = nullTracer
+    , walletEngineTracer = nullTracer
+    , walletDbTracer     = nullTracer
+    , poolsEngineTracer  = nullTracer
+    , poolsDbTracer      = nullTracer
+    , ntpClientTracer    = nullTracer
+    , networkTracer      = nullTracer
+    }
+
+
+{-------------------------------------------------------------------------------
+                                    Logging
+-------------------------------------------------------------------------------}
+
+data StakePoolLog
+    = MsgFollow FollowLog
+    | MsgStartMonitoring [BlockHeader]
+    | MsgHaltMonitoring
+    | MsgCrashMonitoring
+    | MsgRollingBackTo SlotNo
+    | MsgStakePoolGarbageCollection PoolGarbageCollectionInfo
+    | MsgStakePoolRegistration PoolRegistrationCertificate
+    | MsgStakePoolRetirement PoolRetirementCertificate
+    | MsgErrProduction ErrPointAlreadyExists
+    | MsgFetchPoolMetadata StakePoolMetadataFetchLog
+    | MsgFetchTakeBreak Int
+    deriving (Show, Eq)
```

Comment:
Why is the StakePoolLog ending up here with the rest of the application logs? Seems very awkward to me.